packages/editor/src/editor/Editor.tsx

```diff
@@ -55,6 +56,7 @@ const Editor: React.FC<EditorProps> = ({
   lang = languages?.[0].lang ?? 'default',
   pluginsWillChange,
   cellSpacing,
+  uiTranslator = (label) => label,
```

Comment:
as described here https://github.com/react-page/react-page/pull/935#discussion_r606786813

uiTranslator should not have a default `(label) => label`. Instead the {t} function should do that if the user provided uiTranslator does not return a value. This allows us to use default translations.

Reply:
I considered your suggestion, a lot. The t function is being provided via a context and it is a singleton. The same t function is being used everywhere, so it cannot be specific to individual use cases. I don't know how we can implement a useDefaultTranslations function to fit all.

The closest I could think of is that the t function accepts a dictionary as a second argument, and use it in place of useDefaultTranslations. However It gives more problem than it solves and it doesn't work in some cases. For example, in the PluginButton component, it renders all plugin buttons, there is so dynamic, you cannot supply a default dictionary. 

English version labels are being used as key, returning the key as-is covers the lack of translations.

Reply:
here is the t function i proposed:

```
// this is the t function returned by the useUiTranslator hook. 
const t = (key: string) => {
 const translated = userDefinedUiTranslatorFunction(key)
 if(translated === null){
   return defaultTranslatorFunction(key) // that can be the identity to begin with and later can use some default translation object
 }
}
```

`defaultTranslatorFunction would just be `(key) => key`

The problem is, that if we introduce some new label and the developers use an approach where they would not handle fallbacks, we end up with an empty label, which might affect usability negativly.

that's why i am all for handling this case on our side as it is super easy to do.

Your current solution does not do that. If the user provides a translator function that returns null for some keys, it will just not show the label at all.

Reply:
If you spin up the dev server and go to i18n demo page, you will see what I mean. I only provided translations toolbar, and I return "label+(to translate)" for everything else to show all labels are being covered.

Reply:
> here is the t function i proposed:
> 
> 
> 
> ```
> 
> // this is the t function returned by the useUiTranslator hook. 
> 
> const t = (key: string) => {
> 
>  const translated = userDefinedUiTranslatorFunction(key)
> 
>  if(translated === null){
> 
>    return defaultTranslatorFunction(key) // that can be the identity to begin with and later can use some default translation object
> 
>  }
> 
> }
> 
> ```
> 
> 
> 
> `defaultTranslatorFunction would just be `(key) => key`
> 
> 
> 
> The problem is, that if we introduce some new label and the developers use an approach where they would not handle fallbacks, we end up with an empty label, which might affect usability negativly.
> 
> 
> 
> that's why i am all for handling this case on our side as it is super easy to do.
> 
> 
> 
> Your current solution does not do that. If the user provides a translator function that returns null for some keys, it will just not show the label at all.
> 
> 
> 
> 

The signature of the uiTranslator prop require the developers to return a string. So they have to return a string, not a null or undefined. 

It is also very easy for them to always return key as-is as fallback. So they never see a blank label.

Reply:
please implement the fallback in the hook function as described above. Typescript signatures are no runtime guarantee.
 Most developers will just do something like:

```
const uiTranslator = key => MY_TRANSLATION_DICT[key]

```

Is there a show stopper from doing it like that? i described it above in multiple places. Should i push a commit with it?

Reply:
Please push a commit. I don't know how to implement your suggestions. I am keen to see how it works.

Reply:
showed on a comment above

Reply:
Oh, I think I understand now, are you saying if the given function returns null or undefined, the hook default to return key?

Reply:
Do you want me to have one last go or you can take care of it from here?

Sorry for the mess..

Reply:
yes! exactly, does this make sense? Sorry for being unclear and thank you for your effort!

Reply:
sure, go on, i will merge asap, rest looks good!